pkg/service/one2onerestore/worker_validate_integration_test.go

```diff
@@ -0,0 +1,341 @@
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			if tc.setIntereptor != nil {
+				tc.setIntereptor()
+			}
+
+			err := w.validateClusters(context.Background(), tc.manifestsProvider(), tc.hostsProvider(), tc.nodeMappingsProvider())
+			if tc.expecterErr && err == nil {
+				t.Fatalf("Expected err, but got nil")
+			}
+			if !tc.expecterErr && err != nil {
+				t.Fatalf("Unexpected err: %v", err)
+			}
+		})
+	}
+}
+
+func testNewWorker(t *testing.T) (*worker, *testutils.HackableRoundTripper) {
+	t.Helper()
+	hrt := testutils.NewHackableRoundTripper(scyllaclient.DefaultTransport())
+	cfg := scyllaclient.TestConfig(testconfig.ManagedSecondClusterHosts(), testutils.AgentAuthToken())
+	cfg.Transport = hrt
+	sc, err := scyllaclient.NewClient(cfg, log.NopLogger)
+	if err != nil {
+		t.Fatalf("new scylla client: %v", err)
+	}
+
+	managerSession := db.CreateManagedClusterSession(t, false, sc, "", "")
+	clusterSession := db.CreateSession(t, sc)
+
+	w := &worker{
+		managerSession: managerSession,
+		client:         sc,
+		clusterSession: clusterSession,
+		logger:         log.NopLogger,
+	}
+	return w, hrt
+}
+
+func newBackupSvc(t *testing.T, mgrSession gocqlx.Session, client *scyllaclient.Client, clusterID uuid.UUID) *backup.Service {
+	t.Helper()
+	svc, err := backup.NewService(
+		mgrSession,
+		defaultBackupTestConfig(),
+		metrics.NewBackupMetrics(),
+		func(_ context.Context, id uuid.UUID) (string, error) {
+			return "test_cluster", nil
+		},
+		func(context.Context, uuid.UUID) (*scyllaclient.Client, error) {
+			return client, nil
+		},
+		func(ctx context.Context, clusterID uuid.UUID, _ ...cluster.SessionConfigOption) (gocqlx.Session, error) {
+			return db.CreateSession(t, client), nil
+		},
+		testhelper.NewTestConfigCacheSvc(t, clusterID, client.Config().Hosts),
+		log.NopLogger,
+	)
+	if err != nil {
+		t.Fatal(err)
+	}
+	return svc
+}
+
+func defaultBackupTestConfig() backup.Config {
+	return backup.Config{
+		DiskSpaceFreeMinPercent:   5,
+		LongPollingTimeoutSeconds: 1,
+		AgeMax:                    24 * time.Hour,
+	}
+}
+
+func runBackup(t *testing.T, backupSvc *backup.Service, clusterID uuid.UUID, props map[string]any) string {
+	t.Helper()
+	testutils.Printf("Run backup with properties: %v", props)
+	ctx := context.Background()
+	taskID := uuid.NewTime()
+	runID := uuid.NewTime()
+
+	rawProps, err := json.Marshal(props)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "marshal properties"))
+	}
+
+	target, err := backupSvc.GetTarget(ctx, clusterID, rawProps)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "generate target"))
+	}
+
+	err = backupSvc.Backup(ctx, clusterID, taskID, runID, target)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "run backup"))
+	}
+
+	pr, err := backupSvc.GetProgress(ctx, clusterID, taskID, runID)
+	if err != nil {
+		t.Fatal(errors.Wrap(err, "get progress"))
+	}
+	return pr.SnapshotTag
+}
+
+func getNodeMappings(t *testing.T, client *scyllaclient.Client) []nodeMapping {
```

Comment:
It wold be good to signalize (either in a comment or function name) that this gets the identity mapping.